src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -342,6 +344,12 @@ EnsureCitusTableCanBeCreated(Oid relationOid)
 	 * will be performed in CreateDistributedTable.
 	 */
 	EnsureRelationKindSupported(relationOid);
+
+	/*
+	 * When coordinator is added to the metadata, or on the workers,
+	 * some of the relations of the coordinator node may/will be shards.
+	 */
+	ErrorIfRelationIsAKnownShard(relationOid);
```

Comment:
This comment looks a bit irrelevant to me. I think it would be better if we could have a more explanatory comment that tells the reader that we don't allow shards to be distributed. We'll throw this error if the user tries to create a distributed/reference table from a shard. It would be nice if we can tell this in the comment.

Reply:
/*
	 * When coordinator is added to the metadata, or on the workers,
	 * some of the relations of the coordinator node may/will be shards.
	 * We disallow creating distributed tables from shard relations, by
	 * erroring out here.
	 */

Reply:
No, we have a separate check for Citus Local Tables.
https://github.com/citusdata/citus/pull/5694#discussion_r802443357

Reply:
Thanks for the clarification!